### DISCUSSION TITLE
How to fix parameter in kernels?

### DISCUSSION THREAD

**Discussion by HanyangHenry-Wang** [cornellius-gp/gpytorch#2238] [Q&A] [answered]
Hi, I have a question about how to set the parameter of the kernel. 
For example, if I know the lengthscale in RBFKernel and I do not need to train the kernel, how can I set the parameter? Thank you for your help!

**Comment by gpleiss** [comment, marked as answer]
There's a few ways to accomplish this, but the easiest way is to set:

```python
# self.covar_module = ScaleKernel(RBFKernel())
self.covar_module.base_kernel.raw_lengthscale.requires_grad_(False)
```

**Reply by HanyangHenry-Wang** [reply]
Thank you, that is very helpful.

**Reply by findoctorlin** [reply]
> There's a few ways to accomplish this, but the easiest way is to set:
> 
> ```python
> # self.covar_module = ScaleKernel(RBFKernel())
> self.covar_module.base_kernel.raw_lengthscale.requires_grad_(False)
> ```

Thank you! Looks like it can make the "lengthscale shape" of kernel being stable during the training, is it true that the final shape of the underlying function is defined by the final kernel we trained? Like if we get a trained kernel after training process, we use the likelihood function to do prediction as shown in the DGP demo code.
